bin/snapshot.rs

```diff
@@ -0,0 +1,81 @@
+use php_parser_rs::{Lexer, Parser};
+use std::env;
+use std::fs::read_dir;
+use std::path::PathBuf;
+
+fn main() {
+    let manifest = PathBuf::from(env::var("CARGO_MANIFEST_DIR").unwrap());
+    let mut entries = read_dir(manifest.join("tests"))
+        .unwrap()
+        .flatten()
+        .map(|entry| entry.path())
+        .filter(|entry| entry.is_dir())
+        .collect::<Vec<PathBuf>>();
+
+    entries.sort();
+
+    let mut content = String::new();
+    content.push_str("/// this file is auto-generated by the build script.\n");
+    content.push_str("/// you should never manually change it.\n\n\n");
+
+    for entry in entries {
+        let code_filename = entry.join("code.php");
+        let ast_filename = entry.join("ast.txt");
+        let lexer_error_filename = entry.join("lexer-error.txt");
+        let parser_error_filename = entry.join("parser-error.txt");
+
+        if !code_filename.exists() {
+            panic!(
```

Comment:
Instead of panicking, let's just output an info/warning line to say no file was found, similar to how PHPUnit do for test cases with no assertions.

Reply:
but it does mean that the rest of the tests can still be regenerated, an empty directory won't get committed into Git so it's a local only thing

Reply:
okay then, i also updated the build to ignore empty dirs.